File: iterableapi/src/main/java/com/iterable/iterableapi/RetryPolicy.java
package com.iterable.iterableapi;


public class RetryPolicy {

    /**
     * Number of consecutive JWT refresh retries the SDK should attempt before disabling JWT refresh attempts altogether.
     */
    int maxRetry;

    /**
     * Configurable duration in milliseconds between JWT refresh retries. Starting point for the retry backoff.
     */
    long retryIntervalMillis;

    /**
     * Linear or Exponential. Determines the backoff pattern to apply between retry attempts.
     */
    RetryPolicy.Type retryBackoff;
    public enum Type {
        LINEAR,
        EXPONENTIAL
    }
    /**
     * @param retryInterval in seconds
     */
    public RetryPolicy(int maxRetry, long retryInterval, RetryPolicy.Type retryBackoff) {
        this.maxRetry = maxRetry;
        this.retryIntervalMillis = retryInterval * 1000L;
        this.retryBackoff = retryBackoff;
    }
}

File: iterableapi/src/main/java/com/iterable/iterableapi/RetryPolicy.kt
package com.iterable.iterableapi

class RetryPolicy(
    /**
     * Number of consecutive JWT refresh retries the SDK should attempt before disabling JWT refresh attempts altogether.
     */
    val maxRetry: Int,
    /**
     * Configurable duration between JWT refresh retries. Starting point for the retry backoff.
     */
    retryInterval: Long,
    /**
     * Linear or Exponential. Determines the backoff pattern to apply between retry attempts.
     */
    val retryBackoff: Type
) {
    val retryInterval: Long = retryInterval * 1000L

    enum class Type {
        LINEAR,
        EXPONENTIAL
    }
}

